cc/tf_bt_utils.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cc/tf_utils.h"
+
+#include "cc/constants.h"
+#include "cc/mcts_player.h"
+#include "tensorflow/core/example/example.pb.h"
+#include "google/cloud/bigtable/table.h"
+
+using google::cloud::bigtable::Table;
+using google::cloud::bigtable::SetCell;
+using google::cloud::bigtable::CreateDefaultDataClient;
+
+namespace minigo {
+namespace tf_utils {
+
+// Fetches the tensorflow Examples from a MctsPlayer.
+// Linked from tf_utils.cc
+std::vector<tensorflow::Example> FetchExamples(const MctsPlayer& player);
+
+
+// Writes a list of tensorflow Example protos to a series of Bigtable rows.
+int WriteTfExamples(Table& table,
+                    const std::string& row_prefix,
+                    int row_cursor,
+                    const std::vector<tensorflow::Example>& examples) {
+  for (const auto& example : examples) {
+    std::string data;
+    example.SerializeToString(&data);
+    char buf[64];
```

Comment:
nit: this doesn't need to be 64 if you only write a 6digit number into it

Reply:
True. To be honest, I'm wondering whether 6 digits is overachieving, too. It only needs to hold the number of examples written out by a single thread worker. Is there a more sensible upper bound?

Reply:
If the number only needs to be unique to a games then 4d would work (max number of moves is 722)
each worker generates ~ 1 moves every 30 seconds but might run for days so <= 100,000

Reply:
Ok. I'll leave it as `%06d`, then, in the interest of not over-optimizing.